google-cloud-error_reporting/lib/google/cloud/error_reporting/service.rb

```diff
@@ -0,0 +1,118 @@
+# Copyright 2017 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0oud
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+require "google/cloud/errors"
+require "google/cloud/error_reporting/version"
+require "google/cloud/error_reporting/v1beta1"
+require "google/gax/errors"
+
+module Google
+  module Cloud
+    module ErrorReporting
+      ##
+      # @private Represents the gRPC Error Reporting service, including all the
+      #   API methods.
+      class Service
+        attr_accessor :project, :credentials, :host, :timeout, :client_config
+
+        ##
+        # Creates a new Service instance.
+        def initialize project, credentials,
+                       host: nil, timeout: nil, client_config: nil
+          @project = project
+          @credentials = credentials
+          @host = host || V1beta1::ReportErrorsServiceClient::SERVICE_ADDRESS
+          @timeout = timeout
+          @client_config = client_config || {}
+        end
+
+        def channel
+          require "grpc"
```

Comment:
The require will load GRPC, and kick off all of its internal initialization. It also makes it so users cannot fork after afterwards. So we usually generally delay the require as much as possible, wherever possible.